src/main/java/seedu/application/logic/commands/UndoCommand.java

```diff
@@ -0,0 +1,35 @@
+package seedu.application.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.application.model.Model.PREDICATE_SHOW_ALL_APPLICATIONS;
+
+import seedu.application.logic.commands.exceptions.CommandException;
+import seedu.application.model.Model;
+
+/**
+ * Reverts the {@code model}'s application book to its previous state.
+ */
+public class UndoCommand extends Command {
+
+    public static final String COMMAND_WORD = "undo";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Restores the previous application book state from its history.\n";
+
+    public static final String MESSAGE_SUCCESS = "Undo successfully!";
```

Comment:
This one too. Would this be better?
```suggestion
    public static final String MESSAGE_SUCCESS = "Undone successfully!";
```